src/provider/view/components/NotificationCard/NotificationCard.tsx

```diff
@@ -4,34 +4,28 @@ import {NotificationTime} from '../NotificationTime/NotificationTime';
 import {Button} from '../Button/Button';
 import {StoredNotification} from '../../../model/StoredNotification';
 import {CloseButton} from '../CloseButton/CloseButton';
-import {RemoveNotifications, ClickButton, ClickNotification} from '../../../store/Actions';
-import {Actionable} from '../../containers/NotificationCenterApp';
+import {RemoveNotifications, ClickButton, ClickNotification, Actionable} from '../../../store/Actions';
 
 interface NotificationCardProps extends Actionable {
     notification: StoredNotification;
 }
 
 export function NotificationCard(props: NotificationCardProps) {
-    const {notification, storeDispatch} = props;
+    const {notification, storeApi} = props;
     const data = notification.notification;
 
     const handleNotificationClose = () => {
-        storeDispatch(new RemoveNotifications([notification]));
+        new RemoveNotifications([notification]).dispatch(storeApi);
     };
 
     const handleButtonClick = (buttonIndex: number) => {
-        // TODO: Have RemoveNotifications dispatched from inside ClickButton [SERVICE-623]
-        storeDispatch(new ClickButton(notification, buttonIndex));
-        storeDispatch(new RemoveNotifications([notification]));
+        new ClickButton(notification, buttonIndex).dispatch(storeApi);
     };
 
     const handleNotificationClick = (event: React.MouseEvent) => {
         event.stopPropagation();
         event.preventDefault();
-
-        // TODO: Have RemoveNotifications dispatched from inside ClickNotification [SERVICE-623]
```

Comment:
Does this PR complete this story, or are there still more changes to be made?

Reply:
Tagged as 623